src/MfaVerifyController.js

```diff
@@ -123,6 +124,20 @@ function (Okta, BaseLoginController, TOTPForm, YubikeyForm, SecurityQuestionForm
         }
       }
 
+      this.listenTo(this.options.appState, 'change:isWaitingForNumberChallenge',
+        function (state, isWaitingForNumberChallenge) {
+          if (isWaitingForNumberChallenge || this.options.appState.get('lastAuthResponse').status === 'SUCCESS') {
```

Comment:
a couple of questions on this line
1. Is this called everytime we are setting `lastAuthResponse`? or only when the value actually changes from false to true? Reason I am asking is that derived properties are calculated every time, so I want to double check when this is called (could be everytime since the property is recalculated)
2. I see that this also checks for SUCCESS, does that mean that, without number challenge, when we approve a push notification, we will hide the elements? If so, is that needed?

Reply:
1) No. The function is executed only when the value of isWaitingForNumberChallenge changes. ie, if polling doesn't return number challenge the value of isWaitingForNumberChallenge remains false and the function never gets executed. 

2)  The success check will kick in only in number challenge flow and we wont hide other elements in non number challenge flow in success state. 

this is how isWaitingForNumberChallenge  changes in different state 
a) primary auth - false 
b) mfa required - false 
c) mfa challenge  - false 
d) mfa challenge with number challenge data - true
e) success - false ( this is the state that I handle here)

Reply:
1. thanks
2. Got it. So this is a success that will follow showing the number challenge, not any other flow. Makes sense
👍